web_client/actions.jsx

```diff
@@ -35,13 +35,23 @@ export const addAnalysisPage = (page) => promiseAction(
   (D, S) => {
```

Comment:
Might want better names for these function arguments. `D` and `S` are rather opaque, and variables in JavaScript that start with capital letters are supposed to be constructors/classnames.